device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -584,6 +586,7 @@ public void onTransportError(Event event)
     public void onTimerTask(Event event)
     {
         sendQueuedMessages();
+        sendQueuedAcknowledgements();
```

Comment:
does amqp have an ordering requirement that makes this change necessary?

Reply:
No. We can acknowledge messages out of order. This change is just to keep amqp operations in the reactor thread.

Reply:
Thats good, the only other way to handle this is with a uber work queue that just takes things in order. But that might require a bit more refactoring than is necessary.